astropy/io/fits/file.py

```diff
@@ -145,6 +145,16 @@ def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
         if mode is None:
             mode = 'readonly'
 
+        # Handle cloud-hosted files using the optional ``fsspec`` dependency
+        if (kwargs.get('use_fsspec') or _requires_fsspec(fileobj)) and mode != "ostream":
+            # Note: we don't use `get_readable_fileobj` as a context manager
+            # because io.fits takes care of closing files itself
+            fileobj = get_readable_fileobj(fileobj,
+                                           encoding="binary",
+                                           use_fsspec=kwargs.get('use_fsspec'),
+                                           fsspec_kwargs=kwargs.get('fsspec_kwargs'),
+                                           close_files=False).__enter__()
+
```

Comment:
Note: `io.fits` only calls `utils.get_readable_fileobj()` when the need to use `fsspec` is established (either because `use_fsspec=True` was passed, or because the path started with `s3://` or `gs://`).  The existing behavior of `io.fits` is otherwise unaltered.